### PR TITLE
feat(website): A11y updates

### PR DESCRIPTION
**Pull request by kendallstrautman**
[Preview](https://vault-git-kswebsite-a11y-updates-hashicorp.vercel.app/)

This PR fixes some baseline a11y issues. Here are a few snapshots of the axe scans on the homepage.

### Before
Screen Shot 2021-06-16 at 9 53 00 AM

### After
Screen Shot 2021-06-16 at 9 52 43 AM

### CODE REVIEW

File: website/components/before-after-diagram/style.module.css [hunk 1]
```diff
@@ -101,8 +101,6 @@
   }
 
   @media (--medium-up) {
-    height: unset;
```
Comment (kendallstrautman) [inline website/components/before-after-diagram/style.module.css:104]: Fixing a visual bug here

File: website/components/footer/style.css [hunk 2]
```diff
@@ -10,14 +10,12 @@
...
   & a {
-    color: black;
-    opacity: 0.5;
-    transition: opacity 0.25s ease;
+    color: var(--gray-3);
     cursor: pointer;
     display: inline-block;
 
     &:hover {
-      opacity: 1;
```
Comment (kendallstrautman) [inline website/components/footer/style.css:20]: This contrast was an issue

File: website/components/before-after-diagram/index.jsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ function BeforeAfterDiagram(props) {
           </span>
           <div>
             {beforeHeadline && (
-              <h3
+              <h2
```
Comment (kendallstrautman) [inline website/components/before-after-diagram/index.jsx:32]: Fixes an instance of incorrect heading nesting.

File: website/components/hcp-callout-section/index.jsx [hunk 4]
```diff
@@ -12,10 +12,11 @@ export default function HcpCalloutSection({
 }) {
   return (
     <div className={styles.hcpCalloutSection} id={id}>
-      <div className={styles.header}>
-        <h2>{header}</h2>
-      </div>
-
+      {header ? (
+        <div className={styles.header}>
+          <h2>{header}</h2>
```
Comment (kendallstrautman) [inline website/components/hcp-callout-section/index.jsx:17]: This was rendering an empty heading element

File: website/components/hcp-callout-section/index.jsx [hunk 5]
```diff
@@ -28,6 +29,7 @@ export default function HcpCalloutSection({
                 <div key={link.text}>
                   <Button
                     title={link.text}
+                    label="Learn more — HCP Vault"
```
Comment (kendallstrautman) [inline website/components/hcp-callout-section/index.jsx:32]: To add more context re: what 'learn more' refers to

File: website/pages/_app.js [hunk 6]
```diff
@@ -33,26 +33,22 @@ export default function App({ Component, pageProps }) {
         image="https://www.vaultproject.io/img/og-image.png"
         icon={[
           {
-            href:
-              'https://www.datocms-assets.com/2885/1597163356-vault-favicon.png?h=16&w=16',
+            href: 'https://www.datocms-assets.com/2885/1597163356-vault-favicon.png?h=16&w=16',
```
Comment (kendallstrautman) [inline website/pages/_app.js:36]: Just formatting updates here

File: website/pages/_document.js [hunk 7]
```diff
@@ -9,7 +9,7 @@ export default class MyDocument extends Document {
 
   render() {
     return (
-      <Html>
+      <Html lang="en">
```
Comment (kendallstrautman) [inline website/pages/_document.js:12]: 😬 most of the docs sites don't have this attribute defined currently

File: website/pages/home/content.json [hunk 8]
```diff
@@ -5,13 +5,13 @@
       "format": "svg"
     },
     "beforeHeadline": "Static Infrastructure",
-    "beforeContent": "Datacenters with inherently high-trust networks with clear network perimeters.\n\n#### Traditional Approach\n\n- High trust networks\n- A clear network perimeter\n- Security enforced by IP Address",
```
Comment (kendallstrautman) [inline website/pages/home/content.json:8]: This was causing some odd behavior where an empty anchor tag was rendered.

File: website/pages/use-cases/secrets-management/index.jsx [hunk 9]
```diff
@@ -54,13 +59,11 @@ export default function SecretsManagmentUseCase({ content }) {
         features={[
           {
             logo: {
-              url:
-                'https://www.datocms-assets.com/2885/1539889072-1524097013-adobe-white-1.svg',
+              url: 'https://www.datocms-assets.com/2885/1539889072-1524097013-adobe-white-1.svg',
               alt: 'Adobe Logo',
             },
             image: {
-              url:
```
Comment (kendallstrautman) [inline website/pages/use-cases/secrets-management/index.jsx:62]: Formatting updates again here

File: website/components/footer/style.css (hunk 2 above)
Comment (jmfury) [reply website/components/footer/style.css:20]: Should we keep the transition still?

File: website/pages/community/index.jsx [hunk 10]
```diff
@@ -5,39 +5,35 @@ import HashiHead from '@hashicorp/react-head'
 
 function CommunityPage() {
   return (
-    <div id="community">
+    <main id="community">
```
Comment (jmfury) [inline website/pages/community/index.jsx:8]: ✨

File: website/components/footer/style.css (hunk 2 above)
Comment (kendallstrautman) [reply website/components/footer/style.css:20]: There is still a transition. It's transitioning to `black`, which is the same affect as before.

File: website/pages/community/index.jsx [hunk 11]
```diff
@@ -5,39 +5,35 @@ import HashiHead from '@hashicorp/react-head'
...
       <HashiHead is={Head} title="Community | Vault by HashiCorp" />
       <SectionHeader
         headline="Community"
         description="Vault is an open source project with a growing community. There are active, dedicated users willing to help you through various mediums."
-        use_h1={true}
+        useH1={true}
```
Comment (mwickett) [inline website/pages/community/index.jsx:13]: Ouch 😬 This is a _super_ old prop from back in the Middleman / Preact days. Good catch.

File: website/components/footer/style.css [hunk 12]
```diff
@@ -10,14 +10,12 @@
   }
 
   & a {
-    color: black;
-    opacity: 0.5;
-    transition: opacity 0.25s ease;
+    color: var(--gray-3);
```
Comment (jmfury) [inline website/components/footer/style.css:13]:
```suggestion
    color: var(--gray-3);
    transition: color 0.25s ease;
```

File: website/components/footer/style.css (hunk 2 above)
Comment (jmfury) [reply website/components/footer/style.css:20]: Ahh - this is what I meant: https://github.com/hashicorp/vault/pull/11869/files/9a5c7f76b215c0fe3fe3db319f056d06b821c5cd#r654545408

**Review by jescalan — APPROVED** [review]
This is very great

**Review by mwickett — APPROVED** [review]
🎸 Wonderful!

**Pull request merged**